components/match2/wikis/arenaofvalor/matchgroup_input_custom.lua

```diff
@@ -0,0 +1,610 @@
+		Variables.varDefault('tournament_liquipediatiertype')
+	)
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return match
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = match.stream or {}
+	match.stream = {
+		stream = Logic.emptyOr(match.stream.stream, Variables.varDefault('stream')),
+		twitch = Logic.emptyOr(match.stream.twitch or match.twitch, Variables.varDefault('twitch')),
+		twitch2 = Logic.emptyOr(match.stream.twitch2 or match.twitch2, Variables.varDefault('twitch2')),
+		nimo = Logic.emptyOr(match.stream.nimo or match.nimo, Variables.varDefault('nimo')),
+		afreeca = Logic.emptyOr(match.stream.afreeca or match.afreeca, Variables.varDefault('afreeca')),
+		afreecatv = Logic.emptyOr(match.stream.afreecatv or match.afreecatv, Variables.varDefault('afreecatv')),
+		dailymotion = Logic.emptyOr(match.stream.dailymotion or match.dailymotion, Variables.varDefault('dailymotion')),
+		douyu = Logic.emptyOr(match.stream.douyu or match.douyu, Variables.varDefault('douyu')),
+		smashcast = Logic.emptyOr(match.stream.smashcast or match.smashcast, Variables.varDefault('smashcast')),
+		youtube = Logic.emptyOr(match.stream.youtube or match.youtube, Variables.varDefault('youtube')),
+		facebook = Logic.emptyOr(match.stream.facebook or match.facebook, Variables.varDefault('facebook')),
+	}
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+
+	match.links = {}
+	local links = match.links
+	if match.reddit then links.reddit = match.reddit end
+
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	match.extradata = {
+		matchsection = Variables.varDefault('matchsection'),
+		comment = match.comment,
+		mvp = match.mvp,
+		mvpteam = match.mvpteam or match.winner
+	}
+	return match
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, _MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, match.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			opponent.score = string.upper(opponent.score or '')
+			if Logic.isNumeric(opponent.score) then
+				opponent.score = tonumber(opponent.score)
+				opponent.status = _STATUS_SCORE
+				isScoreSet = true
+			elseif Table.includes(_ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = _NOT_PLAYED_SCORE
+			end
+
+			-- get players from vars for teams
+			if opponent.type == _TEAM_OPPONENT_TYPE then
+				if not Logic.isEmpty(opponent.name) then
+					match = matchFunctions.getPlayersOfTeam(match, opponentIndex, opponent.name, opponent.players)
+				end
+			elseif opponent.type == 'solo' then
```

Comment:
Use Module:Opponent for the if statements

Reply:
How do apply module calls into it, or because that a local module call is already there, I can just delete off these lines? (it seems to be 387-399 for the whole thing)

Reply:
```suggestion
			if opponent.type == Opponent.team then
				if not Logic.isEmpty(opponent.name) then
					match = matchFunctions.getPlayersOfTeam(match, opponentIndex, opponent.name, opponent.players)
				end
			elseif Opponent.typeIsParty(opponent) then
```